src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,564 @@
+    // - debug log <entry[mycmd].command.name>
+    //
+    // @Usage
+    // Use to create a global slash command with one option.
+    // - ~discordinteraction id:mybot command create name:animal "description:Pick your favorite!" "options:<list_single[<map[type=string;name=animal;description=Your favorite animal]>]>"
+    //
+    // @Usage
+    // Use to edit the permissions of a command.
+    // - ~discordinteraction id:mybot command perms name:mycmd group:<discord[mybot].group[mygroup]> disable_for:<discord[mybot].group[mygroup].role[Muted]> enabled:true
+    //
+    // @Usage
+    // Use to defer and reply to a slash command interaction.
+    // - ~discordinteraction id:mybot defer interaction:<context.interaction>
+    // - ~discordinteraction id:mybot reply interaction:<context.interaction> <context.options.get[hello].if_null[world]>
+    //
+    // @Usage
+    // Use to defer and reply to an interaction ephermally.
+    // - ~discordinteraction id:mybot defer interaction:<context.interaction> ephermal:true
+    // - ~discordinteraction id:mybot reply interaction:<context.interaction> "Shh, don't tell anyone!"
+    //
+    // -->
+
+    public enum DiscordInteractionInstruction { COMMAND, DEFER, REPLY, DELETE }
+    public enum DiscordInteractionCommandInstruction { CREATE, PERMS, DELETE }
+
+    @Override
+    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
+        for (Argument arg : scriptEntry.getProcessedArgs()) {
+            if (!scriptEntry.hasObject("id")
+                    && arg.matchesPrefix("id")) {
+                scriptEntry.addObject("id", new ElementTag(CoreUtilities.toLowerCase(arg.getValue())));
+            }
+            else if (!scriptEntry.hasObject("instruction")
+                    && arg.matchesEnum(DiscordInteractionInstruction.values())) {
+                scriptEntry.addObject("instruction", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("command_instruction")
+                    && arg.matchesEnum(DiscordInteractionCommandInstruction.values())) {
+                scriptEntry.addObject("command_instruction", arg.asElement());
+            }
+            else if (!scriptEntry.hasObject("group")
+                    && arg.matchesPrefix("group")
+                    && arg.matchesArgumentType(DiscordGroupTag.class)) {
+                scriptEntry.addObject("group", arg.asType(DiscordGroupTag.class));
+            }
+            else if (!scriptEntry.hasObject("name")
+                    && arg.matchesPrefix("name")) {
+                scriptEntry.addObject("name", new ElementTag(CoreUtilities.toLowerCase(arg.getValue())));
+            }
+            else if (!scriptEntry.hasObject("description")
+                    && arg.matchesPrefix("description")) {
+                scriptEntry.addObject("description", new ElementTag(arg.getValue()));
```

Comment:
why is this not `arg.asElement()`

Reply:
was originally lowercase, remnant of copypaste from name arg